test/tools/spec-runner/index.js

```diff
@@ -92,109 +93,123 @@ function gatherTestSuites(specPath) {
     );
 }
 
-function parseTopologies(topologies) {
-  if (topologies == null) {
-    return ['replicaset', 'sharded', 'single'];
+function legacyRunOnToRunOnRequirement(runOn) {
+  const runOnRequirement = new Map();
+
+  if (Array.isArray(runOn.topology)) {
+    expect(
+      isSuperset(Object.values(TopologyType), runOn.topology),
+      `${runOn.topology} includes topology not declared in ${TopologyType}`
+    ).to.be.true;
+    runOnRequirement.set('topologies', runOn.topology);
   }
 
-  return topologies;
-}
+  if (runOn.minServerVersion) {
+    runOnRequirement.set('minServerVersion', patchVersion(runOn.minServerVersion));
+  }
 
-function parseRunOn(runOn) {
-  return runOn.map(config => {
-    const topology = parseTopologies(config.topology);
-    const version = [];
-    if (config.minServerVersion) {
-      version.push(`>= ${config.minServerVersion}`);
-    }
+  if (runOn.maxServerVersion) {
+    runOnRequirement.set('maxServerVersion', patchVersion(runOn.maxServerVersion));
+  }
 
-    if (config.maxServerVersion) {
-      version.push(`<= ${config.maxServerVersion}`);
-    }
+  if (typeof runOn.authEnabled === 'boolean') {
+    runOnRequirement.set('auth', runOn.authEnabled);
+  }
 
-    const mongodb = version.join(' ');
-    return {
-      topology,
-      mongodb,
-      authEnabled: !!config.authEnabled,
-      serverless: config.serverless
-    };
-  });
+  if (typeof runOn.serverless === 'string') {
```

Comment:
How strict should we be here?  

We could also check that the `serverless` parameter is one of `'forbid' | 'allow' | 'require'` in addition to checking that it's a string.

Reply:
Topology satisfies will throw if its an unknown string. My intention here is actually that runOn.serverless is not undefined, I could update the if to that.

In fact this translation function could be simplified to only making the authEnabled -> auth translation, since I think that's the only difference. The code was written like this to avoid undefined to be printed by the metamocha_ui filters. but the unified filter doesn't print out the raw spec, it prints a calculated string